tests/test_add_fund_and_claim.cairo

```diff
@@ -0,0 +1,418 @@
+    Serde::serialize(@month_id, ref event_data_design);
+    Serde::serialize(@design_guild, ref event_data_design);
+    Serde::serialize(@design_fund, ref event_data_design);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_design }
+    ]);
+
+    let mut event_data_problem_solving = Default::default();
+    Serde::serialize(@month_id, ref event_data_problem_solving);
+    Serde::serialize(@problem_solving_guild, ref event_data_problem_solving);
+    Serde::serialize(@problem_solving_fund, ref event_data_problem_solving);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_problem_solving }
+    ]);
+
+    let mut event_data_marcom = Default::default();
+    Serde::serialize(@month_id, ref event_data_marcom);
+    Serde::serialize(@marcom_guild, ref event_data_marcom);
+    Serde::serialize(@marcom_fund, ref event_data_marcom);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_marcom }
+    ]);
+
+    let mut event_data_research = Default::default();
+    Serde::serialize(@month_id, ref event_data_research);
+    Serde::serialize(@research_guild, ref event_data_research);
+    Serde::serialize(@research_fund, ref event_data_research);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_research }
+    ]);
+
+}
+
+#[test]
+fn test_add_fund_same_month_new_guild() { 
+    let initial_supply: u256 = 1000 * TOKEN_MULTIPLIER;
+    let usdc = deploy_mockUSDC(initial_supply);
+    let (master_address, salary_address) = deploy_contracts(usdc);
+
+    let salary_dispatcher = ISalaryDispatcher { contract_address: salary_address };
+    let usdc_dispatcher = IERC20Dispatcher { contract_address: usdc };
+
+    let mut guilds: Array<felt252> = ArrayTrait::new();
+    guilds.append('dev');
+    guilds.append('design');
+    guilds.append('problem_solving');
+    guilds.append('marcom');
+
+    let mut amounts1: Array<u256> = ArrayTrait::new();
+    amounts1.append(10 * TOKEN_MULTIPLIER);
+    amounts1.append(20 * TOKEN_MULTIPLIER);
+    amounts1.append(30 * TOKEN_MULTIPLIER);
+    amounts1.append(40 * TOKEN_MULTIPLIER);
+
+
+    let mut guilds2: Array<felt252> = ArrayTrait::new();
+    guilds2.append('research');
+
+    let mut amounts2: Array<u256> = ArrayTrait::new();
+    amounts2.append(50 * TOKEN_MULTIPLIER);
+
+    start_prank(usdc, deployer_addr());
+    usdc_dispatcher.approve(salary_address, 150 * TOKEN_MULTIPLIER);
+    stop_prank(usdc);
+
+    let month_id = 092023;
+    start_prank(salary_address, deployer_addr());
+    salary_dispatcher.add_fund_to_salary_pools(month_id, amounts1, guilds);
+    salary_dispatcher.add_fund_to_salary_pools(month_id, amounts2, guilds2);
+    stop_prank(salary_address);
+
```

Comment:
As above. I'd add some checks if the values were save correctly, for example for the `_salary_pool ` object

Reply:
added in `test_add_fund()`